nav: chonkify primary navigation

JonasBa opened the pull request:
This PR changes the navigation to fit the new chonk style and fixes a minor issues with dot indicator being rendered in the middle of the list for mobile

The indicators not being positioned at the correct place in the current theme
![CleanShot 2025-04-08 at 20 00 45@2x](https://github.com/user-attachments/assets/28ee8d96-ffa1-4ee0-9b71-78b397b586b8)

![CleanShot 2025-04-08 at 20 16 02@2x](https://github.com/user-attachments/assets/1b562a82-aa92-4e8d-953d-d193350f9589)

Mobile 
![CleanShot 2025-04-08 at 20 16 22@2x](https://github.com/user-attachments/assets/220ea6e9-b72e-43ba-a1e6-931d69b2dd0e)

Mobile root
![CleanShot 2025-04-08 at 20 16 28@2x](https://github.com/user-attachments/assets/83670dec-9bb1-41c3-a1bb-1e94719ee5a3)

JonasBa committed:
nav: chonkify nav

JonasBa committed:
nav: chonkify mobile nav

JonasBa commented on static/app/views/nav/primary/components.tsx:
```diff
@@ -229,29 +249,135 @@ const baseNavItemStyles = (p: {isMobile: boolean; theme: Theme}) => css`
+  border-radius: ${p => p.theme.radius.lg};
+`;
+
+const NavLinkIconContainer = withChonk(
```
@malwilley I've decided to use the `withChonk` component switch and reimplement the components, as it would have otherwise been quite involved and probably very messy to handle the two differences (our button uses the Button component while the current one seems to use the native element).

With the wihtChonk switch, there is a bit more separation and between the two implementations which should also make cleanup later much easier as we can just delete the non chonk components

malwilley replied:
Definitely agree with this approach!

TkDodo commented on static/app/views/nav/primary/components.tsx:
```diff
@@ -329,16 +493,34 @@ export const SidebarItemUnreadIndicator = styled('span')`
+  /* TriggerWrap div is getting in the way here */
+  ${p =>
+    p.theme.isChonk &&
+    css`
+      > div,
+      > li {
+        width: 100%;
+      }
```
the coupling to dom structure is pretty rough here 😢

JonasBa replied:
Yeah, I don't like this. I added both because I assume the intermediary div will be removed at some point because it's also resulting in ul > div html structure

TkDodo commented on static/app/views/nav/primary/components.tsx:
```diff
@@ -313,10 +459,28 @@ export const NavButton = styled('button', {
   position: absolute;
-  top: calc(50% - 12px);
-  left: calc(50% + 12px);
+  top: ${p => (p.isMobile ? `8px` : `calc(50% - 5px)`)};
+  left: ${p => (p.isMobile ? '32px' : `calc(50% + 12px)`)};
```
those numbers look pretty magical to me 😅

JonasBa replied:
Egh, unfortunately not trivial as the buttons are a lot different size in chonk vs the old size

malwilley commented on static/app/views/nav/primary/components.tsx:
```diff
@@ -313,10 +459,28 @@ export const NavButton = styled('button', {
+  return <StyledNavButton {...p} />;
+})``;
+
+export const SidebarItemUnreadIndicator = styled('span')<{isMobile: boolean}>`
```
@JonasBa I noticed one difference - this dot is now a bit too low.

Before/after:

![CleanShot 2025-04-09 at 08 55 23@2x](https://github.com/user-attachments/assets/99d50d8a-92ba-4406-9c05-ce762ff656ab)
![CleanShot 2025-04-09 at 08 55 27@2x](https://github.com/user-attachments/assets/f6b51d7c-c229-474e-a2c1-4b653d35d765)

JonasBa replied:
Ah, good catch! Let me move it higher up!

JonasBa replied:
Fixed upstream!

JonasBa committed:
nav: chonkify nav (#89165)

Chonkify secondary nav 

![CleanShot 2025-04-09 at 10 45
58@2x](https://github.com/user-attachments/assets/0e2ec25f-4b55-44e9-930b-c325821e6e59)
![CleanShot 2025-04-09 at 10 45
47@2x](https://github.com/user-attachments/assets/065cfd9d-65dd-44b7-b1b7-8e98d72d7b9e)

Mobile
![CleanShot 2025-04-09 at 10 45
31@2x](https://github.com/user-attachments/assets/0e6776ef-12cc-4641-8142-e20ff8594972)
